### PR TITLE
 Bluetooth: Make use of Z_STRUCT_SECTION_ITERABLE 

### PR DESCRIPTION
**Pull request by Vudentz**

### CODE REVIEW

File: include/bluetooth/l2cap.h [hunk 1]
```diff
@@ -21,7 +21,7 @@
 extern "C" {
 #endif
 
-#include <sys/atomic.h>
+#include <atomic.h>
```
Comment (jhedberg) [inline include/bluetooth/l2cap.h]: Is this a related change?

**Review by jhedberg — APPROVED** [review]
Looks ok, though the atomic.h include seems out of place - should it be a separate commit?

File: include/bluetooth/l2cap.h (hunk 1 above)
Comment (Vudentz) [reply include/bluetooth/l2cap.h]: It was unintended, it should be fixed now.

**Comment by Vudentz** [comment]
retest

**Comment by Vudentz** [comment]
@jhedberg this one seems to have failed with the regression on the gatt test.

**Comment by joerchan** [comment]
Closed and reopened to run shippable again. Gatt test is fixed in master.

**Review by joerchan — APPROVED** [review]
Looks good.
